src/lib.rs

```diff
@@ -890,8 +911,73 @@ extern "C" fn fn_get_attribute_value(
     let cnt = cast_or_ret!(usize from count => CKR_ARGUMENTS_BAD);
     let mut tmpl: &mut [CK_ATTRIBUTE] =
         unsafe { std::slice::from_raw_parts_mut(template, cnt) };
-    ret_to_rv!(token.get_object_attrs(o_handle, &mut tmpl))
+
+    #[cfg(any(feature = "eddsa", feature = "ec_montgomery"))]
+    let ec_point_len = match tmpl.iter().find(|a| a.type_ == CKA_EC_POINT) {
+        Some(a) => {
+            let gconf = global_rlock!(noinitcheck CONFIG);
+            /* enable the whole thing only if we need to convert to backwards
+             * compatible DER encoding */
+            if gconf.conf.ec_point_encoding == config::EcPointEncoding::Der {
+                let buflen =
+                    cast_or_ret!(usize from a.ulValueLen => CKR_ARGUMENTS_BAD);
+                Some(buflen)
+            } else {
+                None
+            }
+        }
+        None => None,
+    };
+
+    let result = ret_to_rv!(token.get_object_attrs(o_handle, &mut tmpl));
+
+    #[cfg(any(feature = "eddsa", feature = "ec_montgomery"))]
+    if let Some(bufsize) = ec_point_len {
+        use ec::{point_buf_to_der, point_len_to_der};
+
+        match tmpl.iter_mut().find(|a| a.type_ == CKA_EC_POINT) {
+            Some(a) => {
+                if a.ulValueLen == CK_UNAVAILABLE_INFORMATION {
+                    /* do not touch this */
+                    return result;
+                }
+                let buflen =
+                    cast_or_ret!(usize from a.ulValueLen => CKR_GENERAL_ERROR);
+                if a.pValue == std::ptr::null_mut() {
+                    let len = point_len_to_der(buflen);
+                    if len != buflen {
+                        a.ulValueLen = cast_or_ret!(CK_ULONG from len);
+                    }
+                } else {
+                    let buf: &mut [u8] = unsafe {
+                        std::slice::from_raw_parts_mut(
+                            a.pValue as *mut u8,
+                            buflen,
+                        )
+                    };
+                    let out = res_or_ret!(point_buf_to_der(buf, bufsize));
+                    if let Some(v) = out {
+                        if v.len() > bufsize {
+                            return CKR_GENERAL_ERROR;
```

Comment:
should not this be BUFFER_TOO_SMALL error?

Reply:
Not really, this means the DER encoding produced something linger than len + 2, and it means there is an internal bug somewhere.

Reply:
The case were the buffer is smaller than needed is handled by the internal token routines as usual, as with any other attribute. The only case we need to handle is if we have to DER encode the attribute and we have a buffer that is keysize < buflen < keysize+2 and point_buf_to_der() handles that case.